book-es/gen_cover.py

```
@@ -0,0 +1,72 @@
+                                    size="1024x1536", quality="high", n=1)
+        data = base64.b64decode(r.data[0].b64_json)
+        open(out, "wb").write(data)
+        return True
+    except Exception as e:
+        print(f"gpt-image-1 unavailable ({e}); falling back to dall-e-3 …")
+        try:
+            r = client.images.generate(model="dall-e-3", prompt=prompt,
+                                        size="1024x1792", quality="hd",
+                                        style="natural", n=1)
+            url = r.data[0].url
+            with urllib.request.urlopen(url, timeout=30) as resp:
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)